smart-contracts/contracts/utils/UnlockOwner.sol

```diff
@@ -0,0 +1,234 @@
+   * the multisig will be entirely renounced from the current contract with no way to add it back
+   */
+  function changeMultisig(address _newMultisig) external {
+    if ( !_isMultisig()) {
+      revert Unauthorized(msg.sender);
+    }
+    multisigAddress = _newMultisig;
+  } 
+
+
+  /**
+   * Calling this function will execute directly a call to Unlock or a proxy upgrade
+   * @notice This function can only be called by the DAO on mainnet
+   * @param callData the encoded bytes should contains both the call data to be executed and 
+   *  the *action* code that follows `_execAction` pattern (see above).
+   * @return the address of the contract where code has been executed
+   */
+  function execDAO (bytes memory callData) external payable returns (address, bytes memory) {
+     if(!_isDAO()) {
+      revert Unauthorized(msg.sender);
+    }
+    // unpack calldata args
+    (uint8 action, bytes memory execCallData) = abi.decode(callData, (uint8, bytes));
+    (address contractCalled, bytes memory returnedData) = _execAction(action, msg.value, execCallData);
+    return (contractCalled, returnedData);
+  }
+  
+  /**
+   * Calling this function will execute directly a call to Unlock or a proxy upgrade
+   * @notice This function can only be called by the multisig specified in the constructor
+   * @param callData the encoded bytes should contains both the call data to be executed and 
+   *  the *action* code that follows `_execAction` pattern (see above).
+   * @return the address of the contract where code has been executed
+   */
+  function execMultisig (bytes memory callData) external payable returns (address, bytes memory) {
+    if(!_isMultisig()) {
+      revert Unauthorized(msg.sender);
+    }
+    // unpack calldata args
+    (uint8 action, bytes memory execCallData) = abi.decode(callData, (uint8, bytes));
+    (address contractCalled, bytes memory returnedData) = _execAction(action, msg.value, execCallData);
+    return (contractCalled, returnedData);
+  }
+
+  /** 
+   * The receiver function as required by the bridge IXReceiver interface.
+   * @notice calls can only be sent by the DAO on mainnet through the bridge
+   * @dev The Connext bridge contract will call this function, that will trigger
+   * a `_execAction` locally
+   */
+  function xReceive(
+    bytes32 transferId,
+    uint256 amount,
+    address currency,
+    address caller, // address of the contract on the origin chain
+    uint32 origin, // 	Domain ID of the origin chain
+    bytes memory callData
+  ) external returns (bytes memory) {
+
+    // only DAO on mainnet through the bridge
+    if(!_isBridgedDAO(origin, caller)) {
+      revert Unauthorized(msg.sender);
+    }
+
+    // unpack calldata args
+    (uint8 action, bytes memory execCallData) = abi.decode(callData, (uint8, bytes));
+
+    (address contractCalled,  bytes memory returnedData) = _execAction(action, 0, execCallData);
```

Comment:
ok so based on today's conversation I think this is where we would want to "store" execCallData and add a time delay during which it can't be executed.
We would need to add a executeQueued call that anyone can trigger to execute a execCallData but only after time has passed. We should also add a replay-prevention mechanism :)